vendors/espressif/boards/esp32/ports/common_io/iot_i2c.c

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2020 Espressif Systems (Shanghai) Co., Ltd.
+ * All rights reserved.
+ *
+ * You may not use this file except in compliance with the terms and
+ * conditions set forth in the accompanying LICENSE.TXT file.
+ *
+ * THESE MATERIALS ARE PROVIDED ON AN "AS IS" BASIS. ESPRESSIF/AMAZON SPECIFICALLY
+ * DISCLAIMS, WITH RESPECT TO THESE MATERIALS, ALL WARRANTIES, EXPRESS,
+ * IMPLIED, OR STATUTORY, INCLUDING THE IMPLIED WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE, AND NON-INFRINGEMENT.
+ */
+
+#include <string.h>
+#include "iot_i2c.h"
+#include "esp_log.h"
+#include "esp_err.h"
+#include "driver/i2c.h"
+#include "board_gpio.h"
+
+static const char *TAG = "esp-hal-i2c";
+
+#define ACK_CHECK_EN 0x1                        /*!< I2C master will check ack from slave*/
+#define ACK_CHECK_DIS 0x0                       /*!< I2C master will not check ack from slave */
+#define ACK_VAL 0x0                             /*!< I2C ack value */
+#define NACK_VAL 0x1                            /*!< I2C nack value */
+#define SET_REGISTER_ADDR 1
+#define SET_BUFFER_WRITE 0
+#define BYTES_TO_WRITE_AT_ONE_TIME 20
+#define BYTES_TO_READ_AT_ONE_TIME 20
+#define TICKS_TO_WAIT 10
+static bool i2c_flag;
+
+typedef struct {
+    IotI2CConfig_t iot_i2c_config;
+    int32_t i2c_port_num;
+    bool driver_installed;
+    void (*func)(IotI2COperationStatus_t arg1, void *arg2);
+    void *arg;
+    void *pvBuffer;
+    size_t xBytes;
+    bool is_channel_busy;
+    bool is_slave_addr_set;
+    bool is_send_start_flag_set;
```

Comment:
As commented, I think is_send_start_flag_set is not needed.

Reply:
Removed the start flag as discussed